runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListEdit/UsersListEditForm.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { FC, ChangeEvent } from 'react';
+
+import { TextField, MenuItem } from '@mui/material';
+
+import useTranslations from '#src-app/hooks/useTranslations';
+import { emailFieldValidator, requiredFieldValidator } from '#src-app/views/users/UsersListView/UsersListEdit/UsersListEdit.utils';
+
+import { UsersListEditFormProps } from './UsersListEdit.types';
+
+const UsersListEditForm: FC<UsersListEditFormProps> = ({
+    user,
+    setUser,
+    validation,
+    setValidation
+}) => {
+    const emailRegex = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,})+$/;
+
+    const { translate } = useTranslations();
+
+    const getEmailValidator = () => !(validation.email) && emailFieldValidator;
+
+    const getLoginValidator = () => !(validation.login) && requiredFieldValidator;
+
+    const handleEmailChange = (event: ChangeEvent<HTMLInputElement>) => {
+        const email = event.target.value;
+        setUser((prevState) => ({ ...prevState, email }));
+        setValidation((prevState) => ({
+            ...prevState,
+            email: (email.trim() !== '') && emailRegex.test(email)
```

Comment:
this will overwrite useremail with undefined

imo (email.trim() !== '') && emailRegex.test(email) should be at the top of function - that's because at line 26 email is set without checking

Reply:
Validation state is defined by UserDataValidation interface and it is object, which contains only boolean values for their keys.